### PR TITLE
kv/client: support reconnect regions when no event received too long

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Solve https://github.com/pingcap/ticdc/issues/1586 (This is the 5.0 PR, mainly for test, will file another master PR later)

NOTE, **need discussion**: 

1. The check only workers after region is initialized, is this reasonable.
2. The reconnect interval (15m) is hard coded, we need to discuss whether to add a config in server config file. 

After discussion

1. Keep check after the region is initialized, because the incremental scan phase could cost long.
2. We don't need to add it in server config file, since the check is only in real time data phase.

### What is changed and how it works?

- Reuse kv event aliveness check in lock resolver tick
- If a region doesn't receive any event for more than `15 minutes`, force the gRPC stream to reconnect
- Only update the kv client v1, kv client v2 will be updated later

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Force the gRPC stream to reconnect if no event (including resolved ts event) is received for more than 15 minutes.

### CODE REVIEW

File: cdc/kv/client.go [hunk 1]
```diff
@@ -898,6 +907,20 @@ func (s *eventFeedSession) partialRegionFeed(
 		return nil
 	}
 
+	if errors.Cause(err) == errReconnect {
+		cancel, ok := s.getStreamCancel(state.sri.rpcCtx.Addr)
+		if ok {
+			// cancel the stream to trigger strem.Recv with context cancel error
+			// Note use context cancel is the only to terminate a gRPC stream
```
Comment (amyangfei) [inline cdc/kv/client.go:914]:
```suggestion
			// Note use context cancel is the only way to terminate a gRPC stream
```

File: scripts/jenkins_ci/integration_test_common.groovy [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CONCURRENT_NUMBER = 6
+CONCURRENT_NUMBER = 8
```
Comment (amyangfei) [inline scripts/jenkins_ci/integration_test_common.groovy:1]: to accelerate integration test

**Comment by leoppro** [comment]
@amyangfei  please resolve the conflicts

**Review by leoppro — APPROVED** [review]

**Pull request merged**